pkg/addon/controllers/addonannotation/controller.go

```
@@ -0,0 +1,204 @@
+func getAddonAnnotations(annotations map[string]string) map[string]string {
+	result := map[string]string{}
+	for k, v := range annotations {
+		if strings.HasPrefix(k, addonv1alpha1.GroupName) {
+			result[k] = v
+		}
+	}
+	return result
+}
+
+// hasAddonAnnotations returns true if any annotation has the addon prefix.
+func hasAddonAnnotations(annotations map[string]string) bool {
```

golangci-lint (2.5.0)

[error] func hasAddonAnnotations is unused

(unused)